pkg/registry/apps/statefulset/strategy.go

```diff
@@ -85,6 +105,12 @@ func (statefulSetStrategy) PrepareForUpdate(ctx context.Context, obj, old runtim
 
 	pod.DropDisabledTemplateFields(&newStatefulSet.Spec.Template, &oldStatefulSet.Spec.Template)
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.MaxUnavailableStatefulSet) && !maxUnavailableInUse(oldStatefulSet) {
```

Comment:
need tests demonstrating this works properly for all combinations of enabled/disabled, old object with feature present/absent, new object with feature present/absent. see various `func TestDrop...` tests as an example

Reply:
@liggitt  added the tests,ptal